src/methods.ts

```diff
@@ -24,48 +24,30 @@ interface ValidateRequestParams {
 const disableCachingForSpaces = ['magicappstore.eth', 'moonbeam-foundation.eth'];
 
 export async function getVp(params: GetVpRequestParams) {
+  if (['1319'].includes(params.network) || disabled.includes(params.space))
+    throw 'something wrong with the strategies';
+
   if (typeof params.snapshot !== 'number') params.snapshot = 'latest';
 
   if (params.snapshot !== 'latest') {
     const currentBlockNum = await getCurrentBlockNum(params.snapshot, params.network);
     params.snapshot = currentBlockNum < params.snapshot ? 'latest' : params.snapshot;
   }
 
-  const key = sha256(JSON.stringify(params));
-  const useCache =
-    redis && params.snapshot !== 'latest' && !disableCachingForSpaces.includes(params.space);
-  if (useCache) {
-    const cache = await redis.hGetAll(`vp:${key}`);
-
-    if (cache?.vp_state) {
-      cache.vp = parseFloat(cache.vp);
-
-      cache.vp_by_strategy = JSON.parse(cache.vp_by_strategy);
-      return { result: cache, cache: true };
-    }
-  }
-
-  if (['1319'].includes(params.network) || disabled.includes(params.space))
-    throw 'something wrong with the strategies';
-
-  const result = await snapshot.utils.getVp(
-    params.address,
-    params.network,
-    params.strategies,
-    params.snapshot,
-    params.space,
-    params.delegation
+  return await cachedVp(
+    sha256(JSON.stringify(params)),
+    async () => {
+      return await snapshot.utils.getVp(
+        params.address,
+        params.network,
+        params.strategies,
+        params.snapshot,
+        params.space,
+        params.delegation
+      );
+    },
+    params.snapshot !== 'latest' && !disableCachingForSpaces.includes(params.space)
```

Comment:
I have never seen our team using callbacks 🙈 especially `less`

should that be fine?

Reply:
It's the most elegant way to do this thing.

We use in the requestDeduplicator